src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ForEachStatement.cs

```diff
@@ -852,6 +754,125 @@ private BoundStatement RewriteMultiDimensionalArrayForEachStatement(BoundForEach
             return result;
         }
 
+        /// <summary>
+        /// Lower a foreach loop that will enumerate a collection via indexing.
+        /// 
+        /// <![CDATA[
+        /// 
+        /// Indexable a = x;
+        /// for (int p = 0; p < a.Length; p = p + 1) {
+        ///     V v = (V)a[p];   /* OR */   (D1 d1, ...) = (V)a[p];
+        ///     // body
+        /// }
+        /// 
+        /// ]]>
+        /// </summary>
+        /// <remarks>
+        /// NOTE: We're assuming that sequence points have already been generated.
+        /// Otherwise, lowering to for-loops would generated spurious ones.
+        /// </remarks>
+        private BoundStatement RewriteForEachStatementAsFor(BoundForEachStatement node, MethodSymbol indexerGet, MethodSymbol lengthGet)
+        {
+            var forEachSyntax = (CommonForEachStatementSyntax)node.Syntax;
+
+            BoundExpression collectionExpression = GetUnconvertedCollectionExpression(node);
+            NamedTypeSymbol collectionType = (NamedTypeSymbol)collectionExpression.Type;
+
+            TypeSymbol intType = _compilation.GetSpecialType(SpecialType.System_Int32);
+            TypeSymbol boolType = _compilation.GetSpecialType(SpecialType.System_Boolean);
+
+            BoundExpression rewrittenExpression = (BoundExpression)Visit(collectionExpression);
+            BoundStatement rewrittenBody = (BoundStatement)Visit(node.Body);
+
+            // Collection a
+            LocalSymbol collectionTemp = _factory.SynthesizedLocal(collectionType, syntax: forEachSyntax, kind: SynthesizedLocalKind.ForEachArray);
```

Comment:
You can probably drop the parameter names.

Reply:
kind is out of order ( the method takes lots of arguments), but the "syntax:" is indeed can be dropped

---
In reply to: [153343781](https://github.com/dotnet/roslyn/pull/23363#discussion_r153343781) [](ancestors = 153343781)